scripts/backup.sh

```
@@ -0,0 +1,244 @@
+
+# Clean up old backups
+cleanup_old_backups() {
+    log INFO "Starting cleanup of backups older than $RETENTION_DAYS days"
+    
+    local deleted_count=0
+    local total_size_freed=0
+    
+    # Find and delete old backup files
+    while IFS= read -r -d '' file; do
+        if [ -f "$file" ]; then
+            local file_size=$(stat -f%z "$file" 2>/dev/null || stat -c%s "$file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)